gravitee-apim-portal-webui-next/projects/gravitee-dashboard/src/lib/components/filter/generic-filter-bar/generic-filter-bar.component.ts

```diff
@@ -18,74 +18,159 @@ import { Component, effect, input, OnInit, output } from '@angular/core';
 import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
 import { MatFormFieldModule } from '@angular/material/form-field';
 import { MatSelectModule } from '@angular/material/select';
+import moment, { Moment } from 'moment';
 import { distinctUntilChanged, Observable } from 'rxjs';
 
 import { FilterName } from '../../widget/model/request/enum/filter-name';
 import { SelectOption } from '../dropdown-search/dropdown-search-overlay/dropdown-search-overlay.component';
 import { DropdownSearchComponent, ResultsLoaderInput, ResultsLoaderOutput } from '../dropdown-search/dropdown-search.component';
+import { TimeframeSelectorComponent } from '../timeframe-selector/timeframe-selector.component';
+import { customTimeFrames, timeFrames } from '../timeframe-selector/utils/timeframe-ranges';
 
 export interface SelectedFilter {
   parentKey: string;
   value: string;
 }
 
 export interface Filter {
-  key: FilterName;
+  key: FilterName | 'period';
   label: string;
   data?: SelectOption[];
   data$?: Observable<SelectOption[]>;
   dataLoader?: (input: ResultsLoaderInput) => Observable<ResultsLoaderOutput>;
 }
 
+export interface TimeframeValue {
+  period: string;
+  from?: Moment | null;
+  to?: Moment | null;
+}
+
 @Component({
   selector: 'gd-generic-filter-bar',
-  imports: [DropdownSearchComponent, AsyncPipe, ReactiveFormsModule, MatFormFieldModule, MatSelectModule],
+  imports: [DropdownSearchComponent, AsyncPipe, ReactiveFormsModule, MatFormFieldModule, MatSelectModule, TimeframeSelectorComponent],
   templateUrl: './generic-filter-bar.component.html',
   styleUrl: './generic-filter-bar.component.scss',
 })
 export class GenericFilterBarComponent implements OnInit {
   filters = input.required<Filter[]>();
   currentSelectedFilters = input.required<SelectedFilter[]>();
+  defaultPeriod = input<string>('1h');
 
   selectedFilters = output<SelectedFilter[]>();
+  refresh = output<void>();
 
-  form = new FormGroup<Record<string, FormControl<string[]>>>({});
+  form = new FormGroup<Record<string, FormControl<string[] | TimeframeValue | null>>>({});
+
+  protected readonly timeFrames = [...timeFrames, ...customTimeFrames];
+  protected readonly customPeriodKey = 'custom';
 
   constructor() {
+    this.form.addControl('period', new FormControl<TimeframeValue | null>({ period: this.defaultPeriod(), from: null, to: null }));
+
     effect(() => {
       this.filters().forEach(filter => {
         if (!this.form.contains(filter.key)) {
-          this.form.addControl(filter.key, new FormControl<string[]>([], { nonNullable: true }));
+          if (filter.key === 'period') {
+            return;
+          } else {
+            this.form.addControl(filter.key, new FormControl<string[]>([], { nonNullable: true }));
+          }
         }
```

Comment:
This can be simplified:
```suggestion
if (!this.form.contains(filter.key) && filter.key != 'period') {
    this.form.addControl(filter.key, new FormControl<string[]>([], { nonNullable: true }));
}
```